src/steamship/agents/tools/video_generation/did_video_generator_tool.py

```diff
@@ -0,0 +1,120 @@
+"""Tool for generating images."""
+from typing import Any, List, Optional, Union
+
+from steamship import Block, Task
+from steamship.agents.schema import AgentContext
+from steamship.agents.tools import VideoGeneratorTool
+from steamship.utils.repl import ToolREPL
+
+DEFAULT_SOURCE_URL = "https://www.steamship.com/images/agents/man-in-suit-midjourney.png"
+
+
+class DIDVideoGeneratorTool(VideoGeneratorTool):
+    """Tool to generate talking avatars from text using D-ID."""
+
+    name: str = "DIDVideoGeneratorTool"
+    human_description: str = "Generates an a video of you speaking a response to a user."
+    agent_description = (
+        "Used to generate a video of you from text. Use if the user has asked for a video response.  "
+        "The input is the text that you want to say. "
+        "The output is the video of you saying it."
+    )
+    generator_plugin_handle: str = "did-video-generator"
+    generator_plugin_config: dict = {}
+
+    source_url: Optional[str] = DEFAULT_SOURCE_URL
+    """The URL of the source image to be animated."""
+
+    stitch: bool = True
+
+    voice_provider: Optional[str] = "microsoft"
+    """The voice provider. Must be either `microsoft` or `amazon`."""
+
+    voice_id: Optional[str] = "en-US-GuyNeural"
```

Comment:
will this eventually be independent, or is this entirely dependent on `voice_provider`?